conformance/conformance_test.go

```diff
@@ -14,70 +14,138 @@ See the License for the specific language governing permissions and
 	if err != nil {
 		t.Fatalf("Error initializing Kubernetes REST client: %v", err)
 	}
 
-	v1alpha2.AddToScheme(client.Scheme())
-	v1beta1.AddToScheme(client.Scheme())
-	v1.AddToScheme(client.Scheme())
+	v1alpha2.AddToScheme(mgrClient.Scheme())
+	v1beta1.AddToScheme(mgrClient.Scheme())
+	gatewayv1.AddToScheme(mgrClient.Scheme())
+
+	// conformance flags
+	supportedFeatures = suite.ParseSupportedFeatures(*flags.SupportedFeatures)
+	exemptFeatures = suite.ParseSupportedFeatures(*flags.ExemptFeatures)
+	skipTests = suite.ParseSkipTests(*flags.SkipTests)
+	namespaceLabels = suite.ParseKeyValuePairs(*flags.NamespaceLabels)
+	namespaceAnnotations = suite.ParseKeyValuePairs(*flags.NamespaceAnnotations)
+	conformanceProfiles = suite.ParseConformanceProfiles(*flags.ConformanceProfiles)
+	if len(conformanceProfiles) == 0 {
+		t.Fatal("conformance profiles need to be given")
+	}
+	mode = *flags.Mode
+	allowCRDsMismatch = *flags.AllowCRDsMismatch
 
-	supportedFeatures := suite.ParseSupportedFeatures(*flags.SupportedFeatures)
-	exemptFeatures := suite.ParseSupportedFeatures(*flags.ExemptFeatures)
-	skipTests := suite.ParseSkipTests(*flags.SkipTests)
-	namespaceLabels := suite.ParseKeyValuePairs(*flags.NamespaceLabels)
-	namespaceAnnotations := suite.ParseKeyValuePairs(*flags.NamespaceAnnotations)
+	implementation, err = suite.ParseImplementation(
+		*flags.ImplementationOrganization,
+		*flags.ImplementationProject,
+		*flags.ImplementationURL,
+		*flags.ImplementationVersion,
+		*flags.ImplementationContact,
+	)
+	if err != nil {
+		t.Fatalf("Error parsing implementation's details: %v", err)
+	}
+	testConformance(t)
+}
 
+func testConformance(t *testing.T) {
 	t.Logf("Running conformance tests with %s GatewayClass\n cleanup: %t\n debug: %t\n enable all features: %t \n supported features: [%v]\n exempt features: [%v]",
 		*flags.GatewayClassName, *flags.CleanupBaseResources, *flags.ShowDebug, *flags.EnableAllSupportedFeatures, *flags.SupportedFeatures, *flags.ExemptFeatures)
 
-	cSuite := suite.New(suite.Options{
-		Client:     client,
-		RestConfig: cfg,
-		// This clientset is needed in addition to the client only because
-		// controller-runtime client doesn't support non CRUD sub-resources yet (https://github.com/kubernetes-sigs/controller-runtime/issues/452).
-		Clientset:                  clientset,
-		GatewayClassName:           *flags.GatewayClassName,
-		Debug:                      *flags.ShowDebug,
-		CleanupBaseResources:       *flags.CleanupBaseResources,
-		SupportedFeatures:          supportedFeatures,
-		ExemptFeatures:             exemptFeatures,
-		EnableAllSupportedFeatures: *flags.EnableAllSupportedFeatures,
-		NamespaceLabels:            namespaceLabels,
-		NamespaceAnnotations:       namespaceAnnotations,
-		SkipTests:                  skipTests,
-		RunTest:                    *flags.RunTest,
-	})
-	cSuite.Setup(t)
+	cSuite, err := suite.NewConformanceTestSuite(
+		suite.ConformanceOptions{
```

Comment:
Is `*flags.RunTest` missing here?

Reply:
added in https://github.com/kubernetes-sigs/gateway-api/pull/2868/commits/1fd59a4a8aa14033b7eb9270b2b7f14dfbd5ff09